server-spi/src/main/java/org/keycloak/models/RealmModel.java

```diff
@@ -396,6 +396,7 @@
     void updateIdentityProvider(IdentityProviderModel identityProvider);
     Set<IdentityProviderMapperModel> getIdentityProviderMappers();
     Set<IdentityProviderMapperModel> getIdentityProviderMappersByAlias(String brokerAlias);
+    Set<IdentityProviderMapperModel> getIdentityProviderMappersByAliasAndMapper(String brokerAlias, String identityProviderMapper);
```

Comment:
This method should be removed from the model. Use `getIdentityProviderMappersByAlias` and filter by provider ID in the only place it is used in (`SAMLIdentityProvider`) instead.

Reply:
I believe that Java code should have as less as possible loops in lists for having better performance.
See my updating code in model\infinispan\src\main\java\org\keycloak\models\cache\infinispan\RealmAdapter.java .
However,if you ensist, I will do the required change.

Reply:
Considering the gain and maintainability, I still believe it would be better not to change `RealmModel`.

Reply:
Ok.I will do it.